blocks/bricks/conv.py

```diff
@@ -157,9 +160,11 @@ def get_dim(self, name):
         if name == 'input_':
             return (self.num_channels,) + self.image_size
         if name == 'output':
-            return ((self.num_filters,) +
-                    self.get_output_shape(self.image_size, self.filter_size,
-                                          self.step, self.border_mode))
+            image_shape = (None, self.num_channels) + self.image_size
+            kernel_shape = ((self.num_filters, self.num_channels) +
+                            self.filter_size)
+            return self.get_output_shape(image_shape, kernel_shape,
+                                         self.border_mode, self.step)[-3:]
```

Comment:
Because the new thing returns a 4-tuple, which can be None. Our interfaces revolve around 3-tuples.